Tests/kaas/plugin/README.md

```diff
@@ -23,17 +23,91 @@
 
    ```
 
-2. Install dependencies:
+2. **Install dependencies:**
 
    ```bash
    (venv) pip install pip-tools
    (venv) pip-compile requirements.in
    (venv) pip-sync requirements.txt
    ```
 
-3. Set environment variables and launch the process:
+3. **Create a Cluster**
+
+   To create a cluster, run:
 
    ```bash
-   (venv) export CLUSTER_PROVIDER="kind"
-   (venv) python run.py
+   (venv) python run_plugin.py create <plugin-name> ../clusterspec.yaml
    ```
+
+## Setting up and running the cluster-stacks plugin
+
+This section covers how to configure and run the `cluster-stacks` plugin to provision Kubernetes clusters and perform conformance tests.
+
+### Requirements for cluster-stacks plugin
+
+In addition to the core requirements, ensure the following are set up:
+
+* An **OpenStack** environment configured and accessible.
+* A `clouds.yaml` file defining OpenStack credentials and endpoints.
+* An installing the envsubst via this command:
+  ```bash
+  GOBIN=/tmp go install github.com/drone/envsubst/v2/cmd/envsubst@latest
+  ```
+
+### Environment Variables
+
+Before running the `cluster-stacks` plugin, you need to set up the following environment variable:
+
+- **GIT_ACCESS_TOKEN**: This token is required for Git operations, especially if your repository is private.
+
+To set the `GIT_ACCESS_TOKEN`, run the following command in your terminal:
+
+```bash
+export GIT_ACCESS_TOKEN=<your-github-token>
+```
+
+### Configuring clusterspec.yaml file
+
+The `clusterspec.yaml` file is used to set parameters for creating a Kubernetes cluster with the `cluster-stacks` plugin. This file allows you to specify details related to the cluster-stack, Git integration, and cluster configurations.
```

Comment:
Ah, okay. Then we need to think about some new config file where will be defined parameters/variables specific to the plugins (kind, cluster-stacks, etc.) or we can say that the user/tester needs to define some env variables before executing the plugin.

Reply:
The plugin's constructor will take the path of a config file, and that config file should contain everything that is specific to the plugin and the test subject (it could in turn refer to other files, if necessary). The path of the config file can be specified per subject in config.toml.

Reply:
I thought `clusterspec.yaml` was the config file, as it was the only argument (called `CLUSTERSPEC_PATH`) you could pass to the plugin. However, it seems that it was changed in yesterday's commits. I will take a closer look into that.

Reply:
It now take an additional configuration file specific to the plugin. I have also added an example file, see `plugin-cluster-stacks-config.yaml`. 

And I have one question regarding this. How should the plugin configuration file look when you want to provision multiple clusters? Do you need a separate configuration file for each cluster, or is a single file sufficient, with the `clusterspec_cluster` variable determining the cluster?

Reply:
In my mind, the plugin will merge the plugin configuration with the clusterspec information.

Reply:
So it should be merged based on the `clusterspec_cluster`. In my mind, the config file for the plugin would then look something like this:

```yaml
current-k8s-release:
  clouds_yaml_path: "~/.config/openstack/clouds.yaml"   # Path to the OpenStack clouds.yaml file
  cs_name: "scs"                                        # Cluster Stack Name
  cs_version: "v1"                                      # Cluster Stack Version
  # Additional configurations
  ...
  ...
  ...

current-k8s-release-1:
  ...
  ...
  ...
  ```